### PR TITLE
Ignore changes if a video id hasn't been set yet.

### PR DESCRIPTION
**Pull request by thepug**
If a video hasn't been saved yet, no need to auto save a draft.

To test:

- open a new draft
- click record video
- click remove video
- [x] there shouldn't be a draft saved
- start a new draft
- record an actual video
- [x] did the draft get saved with the video in it?

- open a new draft
- record and upload a video
- [x] did the draft get saved?
- remove the video
- click the x (save & close)
- [x] did the draft get removed

- open a new draft
- add an attachment
- [x] did the draft save?
- remove the attachment
- click the x
- [x] did the draft get removed?

- test normal draft operations

### CODE REVIEW

**Comment by bago2k4** [comment]
@thepug i think this is missing some commits, the changes are not related to the PR description.
Also if i start recording a video the POST is saved right away, even before i finish recording.

**Comment by thepug** [comment]
@bago2k4 updated

**Comment by thepug** [comment]
[Remove the draft if it contains only an empty video ID (user hits "record" to see what happens, but never records anything)](https://trello.com/c/aRhW4omK/1537-remove-the-draft-if-it-contains-only-an-empty-video-id-user-hits-record-to-see-what-happens-but-never-records-anything)

File: src/oc/web/components/cmail.cljs [hunk 1]
```diff
@@ -365,7 +365,14 @@
                 {:class (when long-tooltip "long-tooltip")}
                 [:button.mlb-reset.close-bt
                   {:on-click #(do
-                                (autosave s)
+                                (if (and (nil? (:video-id cmail-data))
+                                         (empty? (:attachments cmail-data))
+                                         (clojure.string/blank?
+                                          (:headline cmail-data))
+                                         (clojure.string/blank?
+                                          (:body cmail-data)))
```
Comment (bago2k4) [inline src/oc/web/components/cmail.cljs]: I see we are using this check a couple of times, would be good to have a function in a utility namespace to check if a post has something good to be saved for:
- has attachments or
- has a video or
- has a headline or
- has a body

WYT?

We could reuse that also in `oc.web.actions.activity/remove-video`

**Comment by thepug** [comment]
> I see we are using this check a couple of times, would be good to have a function in a utility >namespace to check if a post has something good to be saved for:

> has attachments or
> has a video or
> has a headline or
> has a body
> WYT?

Yeah, I was thinking about doing this anyway.

**Comment by thepug** [comment]
@bago2k4 ok, I think the last commit has it

**Pull request merged**